From 6bd2286582afa1b6bef494759f4389b0b10122c6 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:09:08 -0400
Subject: [tools] perf/tests: Allow reuse of test_file function

Message-id: <1410181752-25631-304-git-send-email-jolsa@redhat.com>
Patchwork-id: 91967
O-Subject: [PATCH RHEL7.1 BZ1134356 303/307] perf tests: Allow reuse of test_file function
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 822c45db6398a69879b0539f0819de02b813493c
Author: Jiri Olsa <jolsa@kernel.org>
Date: Sun May 4 13:51:46 2014 +0200

description
===========
Making the test_file function to be reusable for
new tests coming in following patches.

Also changing the template name of temp files to
"/tmp/perf-test-XXXXXX" to easily identify & blame.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/tests/dso-data.c b/tools/perf/tests/dso-data.c
index 3e6cb17..7384381 100644
--- a/tools/perf/tests/dso-data.c
+++ b/tools/perf/tests/dso-data.c
@@ -12,11 +12,15 @@
 
 static char *test_file(int size)
 {
- static char buf_templ[] = "/tmp/test-XXXXXX";
+#define TEMPL "/tmp/perf-test-XXXXXX"
+ static char buf_templ[sizeof(TEMPL)];
  char *templ = buf_templ;
  int fd, i;
  unsigned char *buf;
 
+ strcpy(buf_templ, TEMPL);
+#undef TEMPL
+
  fd = mkstemp(templ);
  if (fd < 0) {
   perror("mkstemp failed");
-- 
1.7.1